thousandeyes-sdk-core/test/test_api_client.py

```diff
@@ -0,0 +1,124 @@
+import pytest
+import datetime
+from unittest.mock import Mock, patch
+from thousandeyes_sdk.core.api_client import ApiClient
+from thousandeyes_sdk.core.configuration import Configuration
+from thousandeyes_sdk.core import rest
+from thousandeyes_sdk.core.exceptions import ApiException
+
+
+@pytest.fixture
+def api_client():
+    config = Configuration()
+    return ApiClient(configuration=config)
+
+
+def test_api_client_initialization(api_client):
+    assert api_client.configuration is not None
+    assert isinstance(api_client.rest_client, rest.RESTClientObject)
+    assert api_client.default_headers == {}
+    assert api_client.cookie is None
+
+
+def test_set_default_header(api_client):
+    api_client.set_default_header('X-Test-Header', 'test_value')
+    assert api_client.default_headers['X-Test-Header'] == 'test_value'
+
+
+def test_user_agent_property(api_client):
+    api_client.user_agent = 'test-agent'
+    assert api_client.user_agent == 'test-agent'
+
+
+def test_get_default():
+    default_client = ApiClient.get_default()
+    assert isinstance(default_client, ApiClient)
+
+
+def test_set_default():
+    new_default = ApiClient()
+    ApiClient.set_default(new_default)
+    assert ApiClient.get_default() == new_default
+
+
+def test_param_serialize(api_client):
+    method, url, headers, _, __ = api_client.param_serialize(
+        method='GET',
+        resource_path='/test/{id}',
+        path_params={'id': 1},
+        query_params={'aid': '12'},
+        header_params={'X-Test': 'test_value'}
+    )
+    assert method == 'GET'
+    print(url + "3")
+    assert url == api_client.configuration.host + '/test/1?aid=12'
+    assert headers['X-Test'] == 'test_value'
+
+
+@patch('thousandeyes_sdk.core.rest.RESTClientObject.request')
+def test_call_api(mock_request, api_client):
+    mock_response = Mock()
```

Comment:
i thought that using `@patch` we would need the `Mock()` 🤔  .... but don't take it from me, i'm the worst in python

Reply:
nvm ... 😮‍💨